ecommerce-iosWidget/EcommerceiOSWidget.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  EcommerceiOSWidget.swift
+//  ecommerce-iosWidget
+//
+//  Created by Bliss on 17/5/21.
+//
+
+import WidgetKit
+import SwiftUI
+import Intents
+
+struct Provider: IntentTimelineProvider {
+    
+    func placeholder(in context: Context) -> SimpleEntry {
+        SimpleEntry(date: Date(), configuration: ConfigurationIntent())
+    }
+
+    func getSnapshot(for configuration: ConfigurationIntent, in context: Context, completion: @escaping (SimpleEntry) -> Void) {
+        let entry = SimpleEntry(date: Date(), configuration: configuration)
+        completion(entry)
+    }
+
+    func getTimeline(for configuration: ConfigurationIntent, in context: Context, completion: @escaping (Timeline<Entry>) -> Void) {
+        var entries: [SimpleEntry] = []
+
+        // Generate a timeline consisting of five entries an hour apart, starting from the current date.
+        let currentDate = Date()
+        for hourOffset in 0 ..< 5 {
+            let entryDate = Calendar.current.date(byAdding: .hour, value: hourOffset, to: currentDate)!
```

Comment:
Should we avoid force unwrapping?

Reply:
Nice catch, we won't need this as it's pregen and can be removed.